cockatrice/src/SpoilerBackgroundUpdaterThread.cpp

```diff
@@ -0,0 +1,233 @@
+    runTimer(true);
+}
+
+void SpoilerBackgroundUpdaterThread::runTimer(bool lbStopAndRestart)
+{
+    // Cause a restart of the timer as if we just restart via start() a timeout() will be
+    // thrown which will trigger a download... which we don't want off time!
+    if (lbStopAndRestart)
+    {
+        mpTimerForSpoilers->stop();
+    }
+
+    // Determine how long to wait between updates (Gets the # of hours from settingsCache / convert Mins -> Millis)
+    float lnTimeToWaitBetween = settingsCache->getDownloadSpoilerTimeMinutes() * 60000;
+
+    // How much time is needed until the next update (NOW() - Last Update Time == Time Remaining)
+    long lnLastUpdateTime = settingsCache->getDownloadSpoilerLastUpdateTime();
+    long lnTimeUntilNextUpdate = QDateTime::currentMSecsSinceEpoch() - lnLastUpdateTime;
+    if (lnTimeUntilNextUpdate <= 500 || lnLastUpdateTime == -1)
+    {
+        // The update just occurred less then half a sec ago, so we will restart the timer from the beginning (time zero)
+        mpTimerForSpoilers->start(static_cast<int>(lnTimeToWaitBetween));
+    }
+    else if (lnTimeUntilNextUpdate < lnTimeToWaitBetween)
+    {
+        // If we have checked for an update in the past and it's not yet time to check for another update
+        // Start the clock to count down until lnTimeUntilNextUpdate
+        mpTimerForSpoilers->start(static_cast<int>(lnTimeToWaitBetween - lnTimeUntilNextUpdate));
+    }
+    else
+    {
+        // It's been too long since we last checked (above threshold), so we will start now (time inf)
+        mpTimerForSpoilers->start(0);
+    }
+
+    qDebug() << "Spoiler Timer has started and will finish in" << mpTimerForSpoilers->interval()/60000. << "minutes";
+}
+
+void SpoilerBackgroundUpdaterThread::downloadSpoilersFile(QUrl acUrl)
+{
+    qDebug() << "Starting to download spoiler on NAM";
+    if (mpNetworkAccessManager == nullptr)
+    {
+        mpNetworkAccessManager = new QNetworkAccessManager(this);
+    }
+
+    QNetworkReply *lpNetworkReply = mpNetworkAccessManager->get(QNetworkRequest(acUrl));
+
+    connect(lpNetworkReply, SIGNAL(finished()), this, SLOT(actDownloadFinishedTokensFile()));
+}
+
+void SpoilerBackgroundUpdaterThread::downloadSpoilerSeasonStatusFile()
+{
+    qDebug() << "Checking spoiler season status";
+    if (mpNetworkAccessManager == nullptr)
+    {
+        mpNetworkAccessManager = new QNetworkAccessManager(this);
+    }
+
+    QNetworkReply *lpNetworkReply = mpNetworkAccessManager->get(QNetworkRequest(QString(GITHUB_SPOILERS_STATUS_URL)));
+
+    // Force the calling of actSeeIfSpoilerSeasonIsActive() in sequence instead of parallel
+    // as we need to get that result before we start this timer thread up.
+    QEventLoop lcSubLoop;
+    connect(lpNetworkReply, SIGNAL(finished()), this, SLOT(actSeeIfSpoilerSeasonIsActive()));
+    connect(lpNetworkReply, SIGNAL(finished()), &lcSubLoop, SLOT(quit()));
+    lcSubLoop.exec();
```

Comment:
This 'hack' currently forces the file to be downloaded and parsed _before_ the constructor can continue execution. We need to know if we're in spoiler season or not before continuing, at least that was my idea...